### PR TITLE
[6.10.z] Fix katello-agent tests to use tools repo and register to cdn

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
Cherrypick of https://github.com/SatelliteQE/robottelo/pull/9709 with a few more cherrypicks from master

**Dependant MR**: satlab-tower-MR#562 (merged)

Signed-off-by: Gaurav Talreja <gtalreja@redhat.com>

### CODE REVIEW

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_katello_agent.py

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_katello_agent.py

File: robottelo/hosts.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def __init__(self, hostname, auth=None, **kwargs):
 
         :param hostname: The fqdn of a ContentHost target
         :param auth: ('root', 'rootpass') or '/path/to/keyfile.rsa'
+        :param satellite: optional parameter satellite object.
```
Comment (mshriver) [inline robottelo/hosts.py:125]: When backporting changes like this, its good to maintain the original commit via `cherry-pick`, in order to clearly separate it from other unique changes that you've committed for this branch.

**Comment by mshriver** [comment]
@Gauravtalreja1 what other PRs are these changes coming from? without having the commits split its impossible to tell what is 'new' to this PR and what is already in master and has been reviewed before.

File: robottelo/hosts.py (hunk 1 above)
Comment (Gauravtalreja1) [reply robottelo/hosts.py:125]: @mshriver  To make this PR smaller, I intentionally didn't include original commits via cherry-picks, so I just grabbed changes from `robottelo/hosts.py` to make this tests pass :)
Likewise, if you need related PRs that I referred to - #9633 #9662 #9362 #9310 and the rest are unique changes I've made for this branch.

Comment (mshriver) [reply robottelo/hosts.py:125]: Thanks for the context, this helps. You have to be very careful when not chery-picking the commits, as you might be including function signature updates in `robottelo.hosts` classes without making the requisite updates that were included in the original commit(s).

**Review by pondrejk — COMMENTED** [review]
@Gauravtalreja1 I trust you make sure that tests from cli/test_katello_agent.py are passing.  
I don't see `install_capsule_katello_ca` being used in that test set but it got modified in this PR, so if you could check that any test that actually uses that passes with your changes I would have more confidence to merge this.

File: robottelo/hosts.py [hunk 2]
```diff
@@ -1402,6 +1411,24 @@ def register_to_dogfood(self):
                 f'Error during registration, command output: {cmd_result.stdout}'
             )
 
+    def register_to_cdn(self):
```
Comment (JacobCallahan) [inline robottelo/hosts.py:1414]: This should go under the ContentHost class and raise a ContentHostError instead of CapsuleHostError

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: robottelo/hosts.py (hunk 2 above)
Comment (Gauravtalreja1) [reply robottelo/hosts.py:1414]: @JacobCallahan I agree, ContentHost can also utilize this, updated, please check again, thanks.

**Review by JacobCallahan — APPROVED** [review]

**Comment by Gauravtalreja1** [comment]
@pondrejk Yes, I tested earlier all tests with PRT and they pass, and I don't see any modification happening to `install_capsule_katello_ca`.
Instead, there is an update for `install_katello_ca` and `remove_katello_ca` which is from cherrypick PR #9633 and is used for test_katello_agent, and which works perfectly.

**Pull request merged**